src/server/controllers/movie.js

```diff
@@ -1,31 +1,66 @@
-const jwt = require('jsonwebtoken');
-const { PrismaClient } = require('@prisma/client');
+const jwt = require("jsonwebtoken");
+const { PrismaClient } = require("@prisma/client");
 const prisma = new PrismaClient();
 
-const jwtSecret = 'mysecret';
+const jwtSecret = "mysecret";
 
 const getAllMovies = async (req, res) => {
-    const movies = await prisma.movie.findMany();
+  let payload;
+  try {
+    const [_, token] = req.headers.authorization.split(" ");
+
+    payload = jwt.verify(token, jwtSecret);
+  } catch (e) {
+    return res.status(401).json({ error: "Invalid token provided." });
+  }
+
+  const user = await prisma.user.findFirst({
+    where: { username: payload.username },
+  });
+
+  if (user) {
+    const movies = await prisma.movie.findMany({
+      where: { userId: user.id },
+    });
 
     res.json({ data: movies });
+  }
 };
 
 const createMovie = async (req, res) => {
-    const { title, description, runtimeMins } = req.body;
+  const { title, description, runtimeMins } = req.body;
+  let payload;
 
-    try {
-        const token = null;
-        // todo verify the token
-    } catch (e) {
-        return res.status(401).json({ error: 'Invalid token provided.' })
-    }
+  try {
+    const [_, token] = req.headers.authorization.split(" ");
 
-    const createdMovie = null;
+    payload = jwt.verify(token, jwtSecret);
+  } catch (e) {
+    return res.status(401).json({ error: "Invalid token provided." });
+  }
+
+  const user = await prisma.user.findFirst({
```

Comment:
You should be able to extract these queries to a separate module to clean up this controller and others like it

Your controller gives a function a user, and it expects back either the createdmovie or an error message.

```js
const createMovieFor = (user) => {

// return { error: 'messaged' }
// or return { createdMovie }
}
```